### PR TITLE
Update codecov orb version

### PR DESCRIPTION
**Pull request by tsalo**
Attempting to reproduce errors in #409.

## Changes proposed in this pull request

- Bump codecov orb version from 1.0.5 to 4.1.0.

### CODE REVIEW

**Comment by tsalo** [comment]
First attempt fails with the following:

```
./"codecov"  create-commit -t <redacted>
[221] Error loading Python lib '/tmp/_MEIgcdeCP/libpython3.11.so.1.0': dlopen: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/_MEIgcdeCP/libpython3.11.so.1.0)

Exited with code exit status 255
```

I will add in the following step before `codecov/upload`, based on https://github.com/codecov/codecov-circleci-orb/issues/157#issue-1692263650:

```
- run:
    name: Quick fix for GPG error in Codecov
    command: mkdir -p ~/.gnupg
```

**Comment by tsalo** [comment]
Well this is a new one:

```
./"codecov"  create-commit -t <redacted>
[219] Error loading Python lib '/tmp/_MEICvt1sA/libpython3.11.so.1.0': dlopen: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /tmp/_MEICvt1sA/libpython3.11.so.1.0)

Exited with code exit status 255
```

**Comment by tsalo** [comment]
@mattcieslak I have no clue how, but it's working

File: .circleci/config.yml [hunk 1]
```diff
@@ -594,6 +594,10 @@ jobs:
             coverage xml
       - store_artifacts:
           path: /src/coverage
+      - run:
+          # from https://github.com/codecov/codecov-circleci-orb/issues/157#issue-1692263650
+          name: Quick fix for GPG error in Codecov
+          command: mkdir -p ~/.gnupg
```
Comment (mattcieslak) [inline .circleci/config.yml:600]: huh.

**Review by mattcieslak — APPROVED** [review]

**Pull request merged**